client/lib/src/loginPage.dart

```diff
@@ -1,22 +1,35 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_screenutil/flutter_screenutil.dart';
 import 'package:flushbar/flushbar.dart';
+import 'package:rsa_encrypt/rsa_encrypt.dart';
 import '../main.dart';
 import 'signUpPage.dart';
+import 'dart:async';
+import 'dart:convert';
+import 'package:http/http.dart' as http;
+import './rsa.dart';
+
+// status of any http request
+enum HttpRequestStatus { NOT_DONE, DONE, ERROR }
 
 class LoginPage extends StatefulWidget {
   @override
   LoginPageState createState() => new LoginPageState();
 }
 
 class LoginPageState extends State<LoginPage> {
+  static const _usersUrl = 'http://localhost:8888/users';
+  static final _headers = {'Content-Type': 'application/json'};
+
   final _username = TextEditingController();
   final _password = TextEditingController();
   final _forgottenUsername = TextEditingController();
   bool _validateUsername = false;
   bool _validatePassword = false;
   static bool _validateForgottenUsername = false;
   bool _isSelected = false;
+  bool _inauthEmail = false;
```

Comment:
What are these variables for?

Reply:
inauth variables are for authenticating the user login fields based on the response for the api call.

Reply:
They are also used to show another errorText for the textform fields for username and password, since the current textfield only allows one errortext to show, i have used this logic to showcase the condition that a user enters an invalid email address or password.